File: examples/python/core/invoke/invoke-ondemand-function.py
@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Invoke a mix of durable and on-demand functions."""
    user_id = event["user_id"]
    
    # Invoke a regular Lambda function for data fetching
    user_data = context.invoke(
        function_name="fetch-user-data",  # Regular Lambda function
        payload={"user_id": user_id},
        name="fetch_user",
    )
    
    # Invoke a durable function for complex processing
    processed = context.invoke(
        function_name="process-user-workflow",  # Durable function
        payload=user_data,
        name="process_user",
    )
    
    # Invoke another regular Lambda for notifications
    notification = context.invoke(
        function_name="send-notification",  # Regular Lambda function
        payload={"user_id": user_id, "data": processed},
        name="send_notification",
    )
    
    return {
        "status": "completed",
        "notification_sent": notification["sent"],
    }

File: examples/java/core/invoke/invoke-ondemand-function.java
public class InvokeOnDemandFunctionHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Invoke a mix of durable and on-demand functions
        String userId = (String) event.get("user_id");

        // Invoke a regular Lambda function for data fetching
        var userData = ctx.invoke(
            "fetch_user",
            "fetch-user-data",
            Map.of("user_id", userId),
            Map.class
        );

        // Invoke a durable function for complex processing
        var processed = ctx.invoke(
            "process_user",
            "process-user-workflow",
            userData,
            Map.class
        );

        // Invoke another regular Lambda for notifications
        var notification = ctx.invoke(
            "send_notification",
            "send-notification",
            Map.of("user_id", userId, "data", processed),
            Map.class
        );

        return Map.of(
            "status", "completed",
            "notification_sent", notification.get("sent")
        );
    }
}
